### ISSUE TITLE
[dev/next] Code Review & Cleanup

### ISSUE THREAD

**Issue by TurtleP** [TurtleP/lovebrew#6]
Not my best code for starting a new language, but it works. Either way, this is being done by @ajusa, and I thought an issue would be best to keep track of it. There were some things I wanted us to keep in mind for this:

1. Stop relying on strings
    - It was mentioned to use Enum values instead. 
2. Perhaps change the toml values for the `targets` field to be ctr and hac? Might be too confusing for less technical people though?
3. Whatever I did stupidly about OOP related things to Console/sub-classes should be fixed/removed as needed (i.e. the `method` calls which don't need the `self` variables)
4. 3DS Error Conditions:
    - Asset fails to convert properly
    - Image texture size not within limits [8px, 1024px]

If there's anything I missed, I'll probably add it later on or get feedback from @ajusa.

**Comment by ajusa** [comment]
Additional cleanup:
Unit Tests
Using interfaces instead of subclassing (I think it might make more sense here, especially if we add win/mac/linux down the line)
Using stdlib better in a few places
Including a `nimble` file to make it easier to install all the dependencies for any other contributors
renaming `main.nim` to `lovebrew.nim`

Everything else should be pretty much fine, I'll update this issue with stuff I end up changing. Hoping to get this all done tonight if not tomorrow and send a PR

**Comment by TurtleP** [comment]
I added some error conditions for 3DS -- and while the second point *is* a failure to convert issue, we *need* to ensure the end-user is notified the image was too big or small to convert.

**Comment by ajusa** [comment]
I ran into two rather unfortunate bugs while doing this cleanup and using `iface` - on gcc10 and above, an additional c flag is needed to get it to compile (see first item under https://gcc.gnu.org/gcc-10/porting_to.html). Thankfully, I added a `nim.cfg` file and now it is automatically added. gcc9 and below are unaffected.

Then, I ran into an unusual bug related to `iface` and different modules. This is actually fixed in the devel version of Nim, so I specified that in the `nimble` file. In otherwords, you'll need to do `choosenim devel` which will switch the version of Nim to the development version, since that has a bugfix that is needed.

Thankfully, neither of these affect end users as we just hand them a binary.
